crd-generator/src/main/java/io/strimzi/crdgenerator/CrdGenerator.java

```diff
@@ -427,9 +427,16 @@ private void checkClassOverrides(Class<?> crdClass, String methodName, Class<?>.
         }
         result.putAll(properties(crdClass));
         JsonPropertyOrder order = crdClass.getAnnotation(JsonPropertyOrder.class);
+        if (order == null && !isExemptClass(crdClass)) {
+            throw new InvalidCrdException(crdClass.getName() + " missing @JsonPropertyOrder annotation");
+        }
         return sortedProperties(order != null ? order.value() : null, result).values();
     }
 
+    private boolean isExemptClass(Class<?> crdClass) {
```

Comment:
This method name is too generic. There's plenty of checks on classes in the CrdGenerator, so you need to give it a name which makes clear what it's exempt from. Maybe inverting it and calling it `requiresPropertyOrder(Class<?>)`?